paddle/fluid/imperative/layer.h

```diff
@@ -176,7 +185,8 @@ class OpBase {
       : op_desc_(nullptr),
         forward_id_(-1),
         grad_op_desc_(nullptr),
-        backward_id_(-1) {}
+        backward_id_(-1),
+        place_(platform::CPUPlace()) {}
```

Comment:
not sure we should default to cpu place...

Reply:
Maybe changed to 
```
#ifdef PADDLE_WITH_CUDA
CUDAPlace()
#else 
CPUPlace()
#endif